### PR TITLE
Add codes response

### PR DESCRIPTION
**Pull request by skeleton**

### CODE REVIEW

File: reference/api/index.rst [hunk 1]
```diff
@@ -210,16 +210,101 @@ Create or update a resource
...
-   Location: https://demo.akeneo.com/api/rest/v1/categories/my_category
 
 Let's try our WEB API directly ! (links to RAML definition)
 
 .. toctree::
 
         connector
         rest
+
+Codes Response
```
Comment (a2xchip) [inline reference/api/index.rst]: May be "Response codes" would be better?

File: reference/api/rest.rst [hunk 2]
```diff
@@ -164,11 +164,93 @@ Create or update a resource
...
-For example, after creating a new category with a POST you will get:
+Codes Response
+--------------
+
+Client errors
+~~~~~~~~~~~~~
+
+There are four possible types of client errors on API calls that receive request bodies:
+
+Try to access to the API without authentication will result in a `401 Unauthorized` response.
```
Comment (a2xchip) [inline reference/api/rest.rst]: "Trying"

File: reference/api/rest.rst [hunk 3]
```diff
@@ -164,11 +164,93 @@ Create or update a resource
...
+        "code": 400,
+        "message": "Property 'extra_property' does not exist. Check the standard format documentation.",
+        "_links": {
+            "documentation": {
+                "href": "https://docs.akeneo.com/master/reference/standard_format/other_entities.html#category"
+            }
+        }
+    }
+
+Try to access to a nonexistent resource will result in a `404 Not Found` response.
```
Comment (a2xchip) [inline reference/api/rest.rst]: Same

File: reference/api/rest.rst [hunk 4]
```diff
@@ -164,11 +164,93 @@ Create or update a resource
 When a resource is successfully created or updated, the API returns an HTTP redirection.
 Receiving an HTTP redirection is not an error and clients can follow that redirect if needed.
 
-For example, after creating a new category with a POST you will get:
+Codes Response
```
Comment (a2xchip) [inline reference/api/rest.rst]: Response codes

File: reference/api/rest.rst [hunk 5]
```diff
@@ -164,11 +164,93 @@ Create or update a resource
...
+--------------
+
+Client errors
+~~~~~~~~~~~~~
+
+There are four possible types of client errors on API calls that receive request bodies:
```
Comment (phaseinducer) [inline reference/api/rest.rst]: "that receive request bodies"? I don't understand this part :/

File: reference/api/rest.rst [hunk 6]
```diff
@@ -164,11 +164,93 @@ Create or update a resource
...
+        "errors": [
+            {"field": "code", "message": "This value should not be blank."}
+        ]
+    }
+
+
+Client success
+~~~~~~~~~~~~~~
+
+There are three possible types of client sucess on API calls that receive request bodies:
```
Comment (phaseinducer) [inline reference/api/rest.rst]: idem

File: reference/api/rest.rst [hunk 7]
```diff
@@ -164,11 +164,93 @@ Create or update a resource
...
+
+Get a resource or a collection resources will result in a `200 OK` response.
```
Comment (phaseinducer) [inline reference/api/rest.rst]: getting?

File: reference/api/rest.rst [hunk 8]
```diff
@@ -164,11 +164,93 @@ Create or update a resource
...
+
+Create a resource will result in a `201 Created` response.
+
+.. code-block:: shell
+
+    HTTP/1.1 201 Created
+    Location: https://demo.akeneo.com/api/rest/v1/categories/winter
 
-.. code-block:: bash
+Update a resource will result in a `204 No Content` response.
```
Comment (phaseinducer) [inline reference/api/rest.rst]: updating

**Pull request merged**